bigflow/build/operate.py

```diff
@@ -72,14 +79,64 @@ def _export_docker_image_to_file(tag: str, target_dir: Path, version: str):
     bf_commons.run_process(["docker", "image", "save", "-o", image_target_path, bf_commons.get_docker_image_id(tag)])
 
 
-def _build_docker_image(project_dir: Path, tag: str):
+def _build_docker_image(
+    project_spec: BigflowProjectSpec,
+    tag: str,
+    cache_params: BuildImageCacheParams | None,
+):
+
     logger.debug("Run docker build...")
-    bf_commons.run_process(f'docker build {project_dir} --tag {tag}')
+    cmd = ["docker", "build", project_spec.project_dir, "--tag", tag]
+
+    if cache_params:
+
+        logger.debug("Authenticate to docker registry")
+        bigflow.deploy.authenticate_to_registry(
+            auth_method=cache_params.auth_method or bigflow.deploy.AuthorizationType.LOCAL_ACCOUNT,
+            vault_endpoint=cache_params.vault_endpoint,
+            vault_secret=cache_params.vault_secret,
+        )
+
+        cache_from = _generate_cachefrom_for_docker_build(project_spec, cache_params)
+        logger.debug("Add --cache-from=%s to `docker build`", cache_from)
+        cmd.extend(["--cache-from", cache_from])
+
+
+    # noop when building backend is not a buildkit
+    logger.debug("Enable buildkit inline cache")
+    cmd.extend(["--build-arg", "BUILDKIT_INLINE_CACHE=1"])
+
+    return bf_commons.run_process(cmd)
+
+
+def _generate_cachefrom_for_docker_build(
+    project_spec: BigflowProjectSpec,
+    cache_params: BuildImageCacheParams,
+) -> str:
+    image = cache_params.cache_from_image or ""
+    if cache_params.cache_from_version:
+        image += f",{project_spec.docker_repository}:{cache_params.cache_from_version}"
+        image = image.lstrip(",")
+    return image
+
+
+@dataclass()
+class BuildImageCacheParams:
```

Comment:
It would be nice to have static factories because if you don't know bigflow, you need to know that only specific combinations of these params make sense.

For example:

```
def cache_local(version_or_image: str) -> BuildImageCacheParams: ...
def cache_google_container_registry(version_or_image: str, vault_endpoint: str, vault_secret: str) -> BuildImageCacheParams: ...
```

Reply:
IMO such factories doesn't make big sense because they are not improving our CLI.
Adding them just add some `if .. else` branching into `cli.py` (the only place where `BuildImageCacheParams` is constructed).

Also note that now all options may be passed to `bigflow build` and unused are just ignored (like `--vault-endpoint` may be always passed even when auth type is "LOCAL").